### PR TITLE
feat: Add infra for cleaning up providers when deleted

### LINKED ISSUE #392: Deleting a provider does not clean up the changes on the infrastructure
**Is your feature request related to a problem? Please describe.**
The delete provider option does not undo the changes or recommends you info about how to undo them. Say you add subscriptions to your sns topics, they will stay there and keep sending data to keep.

**Describe the solution you'd like**
Either clean up the changes done, or recommend actions to clean up

### PR DESCRIPTION
**Pull request by skynetigor**
Closes #392

## 📑 Description
Added base functionality for cleaning up provider leftovers when it's deleted in Keep.
Now base provider has clean_up function that each inheritor can override in order to implement its own clean up logic. The function is called during provider deletion.

Add cleaning up webhooks for PagerDuty provider.

## ✅ Checks

- [ ] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [ ] All the tests have passed

### CODE REVIEW

File: keep/providers/providers_service.py [hunk 1]
```diff
@@ -227,34 +227,49 @@ def update_provider(
...
+        try:
+            provider.clean_up()
+        except NotImplementedError:
+            logger.info(
+                "Being deleted provider of type %s does not have a clean_up method",
+                provider_model.type
+            )
+        except Exception as exc:
+            logger.exception(msg="Failed to clean up provider",
+                             extra={ "error_msg": str(exc) })
```
Comment (talboren) [inline keep/providers/providers_service.py:270]: Nice PR! actually `logger.exception` already logs the exception itself alone so no need for that extra param!

Comment (skynetigor) [reply keep/providers/providers_service.py:270]: Nice!
I'll fix,

Comment (skynetigor) [reply keep/providers/providers_service.py]: Fixed :)

**Review by talboren — APPROVED** [review]
LGTM!

**Pull request merged**